py_ecc/curve_properties.py

```diff
@@ -0,0 +1,163 @@
+from typing import (
+    cast,
+)
+
+from py_ecc.field_elements import (
+    FQ,
+    FQ2,
+    FQ12,
+)
+
+from py_ecc.optimized_field_elements import (
+    FQ as optimized_FQ,
+    FQ2 as optimized_FQ2,
+    FQ12 as optimized_FQ12,
+)
+
+from py_ecc.new_typing import (
+    Point2D,
+)
+
+
+curve_properties = {
```

Comment:
I think that this would be better done using a `typing.NamedTuple` so that each of these properties can have stronger types.